src/scala/microsoft-spark-2.3.x/src/main/scala/org/apache/spark/deploy/dotnet/DotnetRunner.scala

```diff
@@ -178,16 +178,20 @@ object DotnetRunner extends Logging {
   // permission to executable (only for Unix systems, since the zip file may have been
   // created under Windows. Finally, the absolute path for the executable is returned.
   private def resolveDotnetExecutable(dir: File, dotnetExecutable: String): String = {
-    val resolvedExecutable = Files
-      .walk(FileSystems.getDefault.getPath(dir.getAbsolutePath))
-      .iterator()
-      .asScala
-      .find(path => Files.isRegularFile(path) && path.getFileName.toString == dotnetExecutable) match {
-      case Some(path) => path.toAbsolutePath.toString
-      case None =>
-        throw new IllegalArgumentException(
-          s"Failed to find $dotnetExecutable under" +
-            s" ${dir.getAbsolutePath}")
+    val resolvedExecutable = if (Files.isRegularFile(Paths.get(dir.getAbsolutePath, dotnetExecutable))) {
+      new File(dir, dotnetExecutable).getAbsolutePath
+    } else {
```

Comment:
Would the following work?
```scala
val path = Paths.get(dir.getAbsolutePath, dotnetExecutable)
val resolvedExecutable = if (Files.isRegularFile(path)) {
  path.toAbsolutePath.toString
} else {
  // `dotnetExecutable` may exist under a subdirectory of `dir` and if the user didn't provide the subdirectory
  // info, the following iterates all files under `dir` to find a matching one.
  ...
}
```

Reply:
Yes, this works. I have changed and let me know if any feedback.